src/components/Landing.js

```diff
@@ -365,21 +368,53 @@ export default class Landing extends Component {
     if (!overviewSection) {
       return false;
     }
-    let stats = [];
+    let stats = {};
     let config = overviewSection.statsConfig;
     if (config) {
       let dataSource = summary[config.dataKeySource] ? summary[config.dataKeySource][config.dataKey]: null;
       let statsSource = dataSource ? dataSource : [];
       if (config.data) {
         config.data.forEach(item => {
-          let o = statsSource;
-          if (item.keyMatch) {
+          let o = statsSource, dataSet = [], statItem = {};
+          let keyMatch = item.keyMatch, summaryFields = item.summaryFields, matchSet = item.matchSet;
+          if (keyMatch && matchSet) {
+            (matchSet).forEach(subitem => {
+              let matchItem = {};
+              matchItem[keyMatch] = subitem.display_name;
+              matchItem.data = [];
+              /* get matching data for each key */
+              subitem.keys.forEach(key => {
+                let matchedData = o.filter(d => {
+                  if (Array.isArray(d[keyMatch])) {
+                    return d[keyMatch].indexOf(key) !== -1;
+                  }
+                  return d[keyMatch] === key;
+                });
+                matchItem.data = [...matchItem.data, ...matchedData];
+              });
+              summaryFields.forEach(f => {
+                if (f.key === "total") {
+                  matchItem[f.key] = matchItem.data.length;
+                  return true;
+                }
+                if (f.identifier) return true;
+                let sd = (matchItem.data).filter(d => d[f.key]);
+                matchItem[f.key] = (Array.from(new Set(sd.map(c => c[f.key])))).length;
+              });
+              dataSet.push(matchItem);
+            });
+            statItem = {
+              fields : summaryFields,
+              data: dataSet
+            };
+            stats[item.title] = statItem;
+          } //end if keyMatch & matchSet
+          else if (item.keyMatch) {
```

Comment:
If I understand correctly, you could reuse the variable you assigned earlier.
```suggestion
          else if (keyMatch) {
```